metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/sql/SequenceGeneration.java

```diff
@@ -0,0 +1,121 @@
+                    new SqlParameterValue(Types.VARCHAR, SEQUENCE_NAME_SERDES),
+                    new SqlParameterValue(Types.VARCHAR, SEQUENCE_NAME_SDS), },
+                (RowCallbackHandler) rs -> {
+                    final String sequenceName = rs.getString("sequence_name");
+                    final Long nextVal = rs.getLong("next_val");
+                    switch (sequenceName) {
+                    case SEQUENCE_NAME_PARTITION:
+                        result.setPartId(nextVal);
+                        break;
+                    case SEQUENCE_NAME_SDS:
+                        result.setSdsId(nextVal);
+                        break;
+                    case SEQUENCE_NAME_SERDES:
+                        result.setSerdeId(nextVal);
+                        break;
+                    default:
+                        throw new IllegalStateException("Failed retrieving the sequence numbers.");
+                    }
+                });
+        } catch (EmptyResultDataAccessException e) {
+            log.warn("Failed getting the sequence ids for partition", e);
+        } catch (Exception e) {
+            throw new IllegalStateException("Failed retrieving the sequence numbers.");
+        }
+
+        try {
+            final List<Object[]> insertValues = Lists.newArrayList();
+            final List<Object[]> updateValues = Lists.newArrayList();
+            if (result.getPartId() == null) {
+                result.setPartId(1L);
+                insertValues.add(new Object[] {result.getPartId() + size, SEQUENCE_NAME_PARTITION });
+            } else {
+                updateValues.add(new Object[] {result.getPartId() + size, SEQUENCE_NAME_PARTITION });
+            }
+            if (result.getSdsId() == null) {
+                result.setSdsId(1L);
+                insertValues.add(new Object[] {result.getSdsId() + size, SEQUENCE_NAME_SDS });
+            } else {
+                updateValues.add(new Object[] {result.getSdsId() + size, SEQUENCE_NAME_SDS });
+            }
+            if (result.getSerdeId() == null) {
+                result.setSerdeId(1L);
+                insertValues.add(new Object[] {result.getSerdeId() + size, SEQUENCE_NAME_SERDES });
+            } else {
+                updateValues.add(new Object[] {result.getSerdeId() + size, SEQUENCE_NAME_SERDES });
+            }
+            if (!insertValues.isEmpty()) {
+                jdbcTemplate.batchUpdate(SQL.SEQUENCE_INSERT_VAL, insertValues,
+                    new int[] {Types.BIGINT, Types.VARCHAR });
+            }
+            if (!updateValues.isEmpty()) {
+                jdbcTemplate.batchUpdate(SQL.SEQUENCE_UPDATE_VAL, updateValues,
+                    new int[] {Types.BIGINT, Types.VARCHAR });
+            }
+            return result;
+        } catch (Exception e) {
+            throw new ConnectorException("Failed updating the sequence ids for partition", e);
+        }
+    }
+
+    @VisibleForTesting
```

Comment:
Instead of using @VisibleForTesting and hidden them into individual class. Another way is to organize these SQL constants into one constant class, and naming explicitly as PART_SQL xxx, SEQ_SQL, xx, where we can get all of them.